libnavigation-copilot/src/main/java/com/mapbox/navigation/copilot/MapboxCopilotImpl.kt

```diff
@@ -360,9 +394,30 @@ internal class MapboxCopilotImpl(
         return context.applicationInfo.flags and ApplicationInfo.FLAG_DEBUGGABLE != 0
     }
 
-    private suspend fun pushHistoryFile(historyFilePath: String, drive: CopilotMetadata) {
+    private fun limitTotalHistoryFilesSize(
+        historyFiles: MutableMap<File, CopilotMetadata>,
+        minFilesCount: Int,
+    ) {
+        var total = historyFiles.keys.sumOf { HistoryAttachmentsUtils.size(it) }
+        while (total > maxTotalHistoryFilesSizePerSession && historyFiles.size > minFilesCount) {
+            val firstFile = historyFiles.keys.first()
```

Comment:
Is the order preserved here?

Reply:
Yes, `mutableMapOf` creates a map that uses insertion-oder for iteration. It is actually a `LinkedHashMap`. Also `toMutableMap` preserves the iteration order.